diff --git a/gulpfile.mjs b/gulpfile.mjs
index 9556eb795..43cc5dac7 100644
--- a/gulpfile.mjs
+++ b/gulpfile.mjs
@@ -16,11 +16,11 @@ gulp.task('copy:backend:views', () =>
 );
 
 gulp.task('copy:frontend:fonts', () =>
-	gulp.src('./packages/frontend/node_modules/three/examples/fonts/**/*').pipe(gulp.dest('./built/_frontend_dist_/fonts/'))
+	gulp.src('./packages/frontend/node_modules/three/examples/fonts/**/*').pipe(gulp.dest('./built/_frontend_dist_/fonts/', { mode: 0o644 }))
 );
 
 gulp.task('copy:frontend:tabler-icons', () =>
-	gulp.src('./packages/frontend/node_modules/@tabler/icons-webfont/**/*').pipe(gulp.dest('./built/_frontend_dist_/tabler-icons/'))
+	gulp.src('./packages/frontend/node_modules/@tabler/icons-webfont/**/*').pipe(gulp.dest('./built/_frontend_dist_/tabler-icons/', { mode: 0o644 }))
 );
 
 gulp.task('copy:frontend:locales', cb => {
